### ISSUE TITLE
Method parameters incorrectly flagged as ignored.

### ISSUE THREAD

**Issue by hannahchan** [SonarSource/sonar-dotnet#3402]
I'm using `SonarAnalyzer.CSharp`, version `8.8.0.18411`.

In the private method of the following code, the parameters of the method get flagged with `Remove this parameter 'propertyName', whose value is ignored in the method. (S1172)` and `Remove this parameter 'descending', whose value is ignored in the method. (S1172)`.

However those values are not ignored and are instead use as part of a tuple.

```C#
namespace FalsePositive
{
    using System.Collections.Generic;
    using System.Collections.Immutable;

    public class SortOrder
    {
        private readonly IImmutableList<(string PropertyName, bool Descending)> sortOrder;

        private SortOrder(IImmutableList<(string PropertyName, bool Descending)> sortOrder)
        {
            this.sortOrder = sortOrder;
        }

        public int Count => this.sortOrder.Count;

        public (string PropertyName, bool Descending) this[int index]
        {
            get => this.sortOrder[index];
        }

        public static SortOrder OrderBy(string propertyName)
        {
            return new SortOrder(ImmutableArray<(string PropertyName, bool Descending)>.Empty)
                .Add(propertyName, false);
        }

        public static SortOrder OrderByDescending(string propertyName)
        {
            return new SortOrder(ImmutableArray<(string PropertyName, bool Descending)>.Empty)
                .Add(propertyName, true);
        }

        public SortOrder ThenBy(string propertyName)
        {
            return this.Add(propertyName, false);
        }

        public SortOrder ThenByDescending(string propertyName)
        {
            return this.Add(propertyName, true);
        }

        public IEnumerator<(string PropertyName, bool Descending)> GetEnumerator()
        {
            return this.sortOrder.GetEnumerator();
        }

        private SortOrder Add(string propertyName, bool descending)
        {
            return new SortOrder(this.sortOrder.Add((propertyName, descending)));
        }
    }
}
```

**Comment by pavel-mikula-sonarsource** [comment]
Hi @hannahchan,

Thank you for reporting this issue. I can confirm it's a False Positive. We already have similar case reported in #3255 with the same root cause so I'll close this one.
